src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -324,6 +324,12 @@ def forEachVar (hs : Array Syntax) (tac : MVarId → FVarId → MetaM MVarId) :
 @[builtin_tactic Lean.Parser.Tactic.substVars] def evalSubstVars : Tactic := fun _ =>
   liftMetaTactic fun mvarId => return [← substVars mvarId]
 
+/--
+`subst_eq` repeatedly substitutes according to the equality proof hypotheses in the context,
+replacing the left side of the equality with the right, until no more progress can be made.
+-/
```

Comment:
This docstring isn't accurate. The `Lean.MVarId.substEqs` function will try doing `cases` on every `Eq` and `HEq` in the local context. That means it even will succeed in cases like `Nat.succ a = Nat.succ b`, substituting using `a = b`.

It doesn't have any preference for the LHS or the RHS -- if either side is a variable, it will be substituted.